client/sections-middleware.js

```diff
@@ -0,0 +1,179 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import config from 'config';
+import page from 'page';
+import { find, groupBy, toPairs, partial, escapeRegExp } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { activateNextLayoutFocus } from 'state/ui/layout-focus/actions';
+import utils from '../server/bundler/utils';
+import * as LoadingError from 'layout/error';
+import * as controller from './controller/index.web';
+import { restoreLastSession } from 'lib/restore-last-path';
+import { hub as preloadHub } from 'sections-preload';
+import { switchCSS } from 'lib/i18n-utils/switch-locale';
+
+import sections from './sections';
+
+const sectionsWithCss = sections.map( section => ( {
+	...section,
+	...( section.css && {
+		css: {
+			id: section.css,
+			urls: utils.getCssUrls( section.css ),
+		},
+	} ),
+} ) );
+
+const _loadedSections = {};
+
+function maybeLoadCSS( sectionName ) {
+	//eslint-disable-line no-unused-vars
+	const section = find( sectionsWithCss, elem => elem.name === sectionName );
+
+	if ( ! ( section && section.css ) ) {
+		return;
+	}
+
+	const url =
+		typeof document !== 'undefined' && document.documentElement.dir === 'rtl'
+			? section.css.urls.rtl
+			: section.css.urls.ltr;
+
+	switchCSS( 'section-css-' + section.css.id, url );
+}
+
+function preload( sectionName ) {
+	maybeLoadCSS( sectionName );
+	const section = find( sections, { name: sectionName } );
+	if ( ! section ) {
+		return Promise.reject( `Attempting to load non-existent section: ${ sectionName }` );
+	}
+
+	// if ( async ) {
+	const chunk = moduleToImport[ section.module ];
+	return chunk;
+	// }
+
+	// @TODO: get non-async mode working
+	// return require( section.module );
+}
+
+const moduleToImport = {
+	'jetpack-connect': import( /* webpackChunkName: "'jetpack-connect'" */ 'jetpack-connect' ),
```

Comment:
What happens here is once the browser hit that, it'll fetch all sections, that's now how the loader used to work